airflow/www/auth.py

```diff
@@ -32,9 +32,6 @@ def decorated(*args, **kwargs):
             appbuilder = current_app.appbuilder
             if appbuilder.sm.check_authorization(permissions, request.args.get('dag_id', None)):
                 return func(*args, **kwargs)
-            else:
-                access_denied = "Access is Denied"
-                flash(access_denied, "danger")
```

Comment:
So, this isn't quite right.

This helps when the user isn't logged in, but if you are a non-admin user, and you, for example, try to go to `http://localhost:8080/connection/list/` then it will redirect you (in a _very_ round about way via /login -> /) to /home and show this "access is deined" flash.

I think without that flash message it would be very confusing.

So we probably need to do something like looking at current_user.is_anonymous (I think that's the property) -- if it's anon, don't show the flash, but if there is a user then, yes, you can't actually access this page.

Reply:
@ashb ok I updated it. I realized a lot of CRUD paths use the built-in FAB routing and access controls and never even use this function. So instead I just modified `flash.html` to hide this specific error on the login page. Not the prettiest solution, but at least it's consistent.